### PR TITLE
TITLE: UofT-DSI | LCR- Assignment-1

### PR DESCRIPTION
**Pull request by minamahdian**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
I did data exploration and implemented KNN. 
## What did you learn from the changes you have made?
I learned to use a cross-validation set to improve model accuracy and a seed to enhance the reproducibility of predictions.
## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
I can use cross-validation and hyperparameter tuning.
## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
N/A
## How were these changes tested?
I tested my code in Jupyter notebook. 
## A reference to a related issue in your repository (if applicable)
N/A
## Checklist
- [ x] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by KasraVakiloroayaei — COMMENTED** [review]
A1: Missing

**Review by KasraVakiloroayaei — CHANGES_REQUESTED** [review]
A1: Incomplete

Q3:Define the parameter grid for n_neighbors using the following approach :
parameter_grid = {
    "n_neighbors": range(1, 51),
}

Q4:Recalculate accuracy once again after you have obtained the new RMSE
